tf2_ros/include/tf2_ros/transform_listener.h

```diff
@@ -125,43 +126,49 @@ class TransformListener
     callback_t static_cb = std::bind(
       &TransformListener::subscription_callback, this, std::placeholders::_1, true);
 
-    message_subscription_tf_ = rclcpp::create_subscription<tf2_msgs::msg::TFMessage>(
-      node,
-      "/tf",
-      qos,
-      std::move(cb),
-      options);
-    message_subscription_tf_static_ = rclcpp::create_subscription<tf2_msgs::msg::TFMessage>(
-      node,
-      "/tf_static",
-      static_qos,
-      std::move(static_cb),
-      static_options);
-
-    if (spin_thread) {
-      initThread(node->get_node_base_interface());
+    if (spin_thread_) {
+      // Create new callback group for message_subscription of tf and tf_static
+      callback_group_ = node_base_interface_->create_callback_group(
+        rclcpp::CallbackGroupType::MutuallyExclusive, false);
+      // Duplicate to modify option of subscription
+      rclcpp::SubscriptionOptionsWithAllocator<AllocatorT> options2 = options;
+      rclcpp::SubscriptionOptionsWithAllocator<AllocatorT> static_options2 = static_options;
+      options2.callback_group = callback_group_;
+      static_options2.callback_group = callback_group_;
+      message_subscription_tf_ = rclcpp::create_subscription<tf2_msgs::msg::TFMessage>(
+        node, "/tf", qos, std::move(cb), options2);
+      message_subscription_tf_static_ = rclcpp::create_subscription<tf2_msgs::msg::TFMessage>(
+        node, "/tf_static", static_qos, std::move(static_cb), static_options2);
+      // Create executor with dedicated thread to spin.
```

Comment:
A space here might be nice for a logical break between creating the subscribers and then setting up the executor